includes/collection/class-inbox.php

```diff
@@ -140,4 +141,93 @@ public static function get( $guid, $user_id ) {
 
 		return \get_post( $post_id );
 	}
+
+	/**
+	 * Get an inbox item by its GUID.
+	 *
+	 * @param string $guid The GUID of the inbox item.
+	 *
+	 * @return \WP_Post|\WP_Error The inbox item or WP_Error.
+	 */
+	public static function get_by_guid( $guid ) {
+		global $wpdb;
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		$post_id = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT ID FROM $wpdb->posts WHERE guid=%s AND post_type=%s",
+				\esc_url( $guid ),
+				self::POST_TYPE
+			)
+		);
+
+		if ( ! $post_id ) {
+			return new \WP_Error(
+				'activitypub_inbox_item_not_found',
+				\__( 'Inbox item not found', 'activitypub' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		return \get_post( $post_id );
+	}
+
+	/**
+	 * Undo a received activity.
+	 *
+	 * @param string $id The ID of the inbox item to be removed.
+	 *
+	 * @return bool|\WP_Error True on success, WP_Error on failure.
+	 */
+	public static function undo( $id ) {
+		$type = '';
+		$post = self::get_by_guid( $id );
+
+		if ( \is_wp_error( $post ) ) {
+			// If inbox entry not found, return the error.
+			return $post;
+		}
+
+		$type = \get_post_meta( $post->ID, '_activitypub_activity_type', true );
+
+		switch ( $type ) {
+			case 'Follow':
+				$actor   = \get_post_meta( $post->ID, '_activitypub_activity_remote_actor', true );
+				$user_id = $post->post_author;
+				$post    = Remote_Actors::get_by_uri( $actor );
+
+				if ( \is_wp_error( $post ) ) {
+					return $post;
+				}
+
+				return Followers::remove( $post, $user_id );
+			case 'Like':
```

Comment:
How would you feel about a blank line between case blocks to give them more visual separation?